pages/c/_cluster/explorer/index.vue

```diff
@@ -182,29 +372,53 @@ export default {
   },
 
   methods: {
-    showActions() {
-      this.$store.commit('action-menu/show', {
-        resources: this.cluster,
-        elem:      this.$refs['cluster-actions'],
-      });
+    createMemoryValues(total, useful) {
+      const parsedTotal = parseSi((total || '0').toString());
+      const parsedUseful = parseSi((useful || '0').toString());
+      const format = this.createMemoryFormat(parsedTotal);
+      const formattedTotal = formatSi(parsedTotal, format);
+      const formattedUseful = formatSi(parsedUseful, format);
+
+      return {
+        total:  Number.parseFloat(formattedTotal),
+        useful: Number.parseFloat(formattedUseful),
+        units:  this.createMemoryUnits(parsedTotal)
+      };
     },
 
-    async fetchConstraints() {
-      try {
-        const rawConstraints = await findAllConstraints(this.$store);
+    createMemoryFormat(n) {
+      const exponent = exponentNeeded(n, PARSE_RULES.memory.format.increment);
```

Comment:
The only reason to use `exponentNeeded` and then ask for a particular `{min,max}Exponent` is if you have several separate values to show and are trying to ensure they are all going to be displayed in the same unit.  For just one value you can not pass in min/max at all and will get the same result.

Reply:
Maybe I'm misunderstanding but there is more than one number being formatted. The formattedTotal and formattedUseful.

Reply:
But you're calculating and using them independently, so this is the same as not specifying an exponent all.  The only reason to call `exponentNeeded` yourself is for things like used=805306368 total=1073741824 and you want multiple displayed values to be used together and all share the same unit, e.g. `{used} of {total} {unit} => "0.75 of 1 GiB"`.

If you format the 2 numbers independently you'd get "768 MiB" and "1 GiB".  But for this use you want them to be in the same unit, so you find out what exponent you should be using: `exp = Math.max(exponentNeeded(used), exponentNeeded(total))` (which will be `3`) and then format both forcing the exponent to that: `{minExponent: exp, maxExponent: exp}` so that they come out as "0.75 [GiB, though you'd remove this]" and "1 GiB".

Reply:
Ok now I see that the same single `format` is used for two calls instead of createMemoryFormat being called twice.

So this mostly "works", but is implicitly assuming that requested (useful) <= allocatable (total).  Which is not necessarily true; it is possible for a cluster to allow oversubscription.